aleph-client/src/lib.rs

```diff
@@ -51,39 +51,23 @@ pub type Header = GenericHeader<BlockNumber, BlakeTwo256>;
 pub type KeyPair = sr25519::Pair;
 pub type Connection = Api<KeyPair, WsRpcClient>;
 
-#[derive(Copy, Clone, Debug)]
-pub enum Protocol {
-    WS,
-    WSS,
+pub fn create_connection(address: &str) -> Connection {
+    create_custom_connection(address).expect("Connection should be created")
 }
 
-impl Default for Protocol {
-    fn default() -> Self {
-        Protocol::WS
+/// Unless `address` already contains protocol, we prepend to it `ws://`.
+fn ensure_protocol(address: &str) -> String {
+    if address.starts_with("ws://") || address.starts_with("wss://") {
+        return address.to_string();
     }
-}
-
-pub fn from(use_ssl: bool) -> Protocol {
-    match use_ssl {
-        true => Protocol::WSS,
-        false => Protocol::WS,
-    }
-}
-
-pub fn create_connection(address: &str, protocol: Protocol) -> Connection {
-    create_custom_connection(address, protocol).expect("connection should be created")
+    format!("ws://{}", address)
```

Comment:
what about `impl Display for Protocol` and `Protocol::WS.to_string();`

Reply:
You mean reviving `Protocol` enum and use its string representation instead of in-place strings?

Reply:
yes

Reply:
done